Gems/ROS2/Code/Source/Lidar/ROS2Lidar2DSensorComponent.cpp

```diff
@@ -173,12 +172,28 @@ namespace ROS2
         ConnectToLidarRaycaster();
         ConfigureLidarRaycaster();
 
-        ROS2SensorComponent::Activate();
+        m_sourceEventHandler = decltype(m_sourceEventHandler)([this](auto... args)
+        {
+            this->Visualize();
+        });
+        m_eventSourceAdapter.ConnectToSourceEvent(m_sourceEventHandler);
+
+        m_adaptedEventHandler = decltype(m_adaptedEventHandler)([this](auto... args)
+        {
+            this->FrequencyTick();
+        });
+        m_eventSourceAdapter.ConnectToAdaptedEvent(m_adaptedEventHandler);
+
+        m_eventSourceAdapter.Configure(m_sensorConfiguration);
+        m_eventSourceAdapter.Activate();
```

Comment:
There is nothing lidar-specific to this added code block, and it looks like it would be repeated in each sensor implementation (with Tick source). Is it possible to do it in ROS2SensorComponentBase or generalize otherwise?